packages/foam-vscode/src/core/services/markdown-link.ts

```diff
@@ -38,29 +38,42 @@ export abstract class MarkdownLink {
 
   public static createUpdateLinkEdit(
     link: ResourceLink,
-    delta: { target?: string; section?: string; alias?: string }
+    delta: {
+      target?: string;
+      section?: string;
+      alias?: string;
+      type?: 'wikilink' | 'link';
+      isEmbed?: boolean;
+    }
   ) {
     const { target, section, alias } = MarkdownLink.analyzeLink(link);
     const newTarget = delta.target ?? target;
     const newSection = delta.section ?? section ?? '';
     const newAlias = delta.alias ?? alias ?? '';
     const sectionDivider = newSection ? '#' : '';
     const aliasDivider = newAlias ? '|' : '';
-    const embed = link.isEmbed ? '!' : '';
-    if (link.type === 'wikilink') {
+    const embed = (() => {
+      if (typeof delta.isEmbed === 'boolean') {
+        return delta.isEmbed ? '!' : '';
+      }
+      return link.isEmbed ? '!' : '';
+    })();
+    const type = delta.type ?? link.type;
+    const useAngles = newTarget.indexOf(' ') > 0 || newSection.indexOf(' ') > 0;
```

Comment:
minor: I suggest moving this inside the `type === 'link'` block, as it's only necessary there